### PR TITLE
[Dart][Jaguar] and proto serialization support 

### PR DESCRIPTION
**Pull request by jaumard**
Here is my current work to allow the generator of jaguar dart to support either json or proto serialization. 
I've added a `serializarion` param to choose json or proto. 
I proto models should be generated as proto files and not dart files. For now I'm blocked because I'm trying to do this: 
```
message {{classname}} {
    {{#vars}}
    {{{datatype}}} {{name}} = {{@index}};
    {{/vars}}
}
```
Apparently the {{@index}} is not supported :/ and return empty. But in proto definition you need to specify the index of each field like: 
```
message Person {
  string name = 1;
  int32 id = 2;
  string email = 3;

}
```

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/DartJaguarClientCodegen.java [hunk 1]
```diff
@@ -152,6 +194,19 @@ public void processOpts() {
                     modelImports.add(underscore(modelImport));
                 }
             }
+
+            int index = 1;
+            for (CodegenProperty p : cm.vars) {
+                p.vendorExtensions.put("x-index", index);
+                String protoType = protoTypeMapping.get(p.baseType);
+                if(p.isListContainer) {
+                    String innerType = protoTypeMapping.get(p.mostInnerItems.baseType);
```
Comment (jaumard) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/DartJaguarClientCodegen.java]: this is not working yet, as here we have the dart type and not the openapi type :(

Comment (jaumard) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/DartJaguarClientCodegen.java]: This is now ok !

**Comment by jaumard** [comment]
It will need a real test case to really know if it works but this should be quite good. !

**Comment by jaumard** [comment]
@wing328 the WIP label doesn't go away automatically ^^

**Comment by jaumard** [comment]
@wing328 can we merge this ? :) don't want to keep rebase in it forever ^^

**Comment by wing328** [comment]
@jaumard sure. Can you resolve the merge conflicts? Let me know if you need any help.

**Pull request merged**